proposals/buildfile.md

```diff
@@ -0,0 +1,186 @@
+# Jib CLI Buildfile Specification
+
+Specification for a YAML buildfile describing building a container image. This buildfile can be
+used by the jib-cli to generate a container. It is translated directly into a buildplan and
+passed to the builder.
+
+```yaml
+apiVersion: jib/v0alpha1
+kind: Config
+
+# "FROM" with detail for manifest lists or multiple architectures
+from:
+  image: "gcr.io/distroless/java:8"
+  platforms:
+    - architecture: "arm"
+      os: "linux"
+      os.version: "a"
+      os.features:
+        - "b1"
+        - "b2"
+      variant: "c"
+      features:
+        - "d1"
+        - "d2"
+    - architecutre: amd64
+      os: darwin
+
+# potentially simple form of "FROM" (based on ability to define schema)
+from: "gcr.io/distroless/java:8"
+
+creationTime: 0 # millis since epoch or iso8601 creation time
+format: Docker # Docker or OCI
+
+environment:
+  "MY_KEY1": "value"
```

Comment:
I haven't done an exhaustive search but it feels to me most other environments use an array of strings:
```yaml
environment:
  - "MYKEY1=value"
```

That said, map-style allows using aliases and anchors so 🤷

Reply:
Yeah I dunno, I have this parsing correctly like this, and it seems to make sense. I would prefer to keep it this way, unless the ecosystem has decided otherwise.

Reply:
Jib currently uses a map style config. Will leave like this for now.